swift/ql/src/queries/Security/CWE-259/ConstantPassword.qhelp

```diff
@@ -0,0 +1,21 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+  <overview>
+    <p>Deriving password-based encryption keys using hard-coded passwords is insecure, because the generated key may be easily discovered. Data hashed using constant salts are vulnerable to dictionary attacks, enabling attackers to recover the original input.</p>
+  </overview>
+
+  <recommendation>
+    <p>Use randomly generated passwords to securely derive a password-based encryption key.</p>
+  </recommendation>
+
+  <example>
+    <p>The following example shows a few cases of hashing input data. In the 'BAD' cases, the password is constant, making the derived key vulnerable to dictionary attakcs.  In the 'GOOD' cases, the password is randomly generated, which protects the hashed data against recovery.</p>
+    <sample src="ConstantPassword.swift" />
+  </example>
+
+  <references>
+    <li><a href="https://www.okta.com/blog/2019/03/what-are-salted-passwords-and-password-hashing/">What are Salted Passwords and Password Hashing?</a></li>
```

Comment:
[This RFC](https://www.rfc-editor.org/rfc/rfc2898) is more about the process where these passwords are stored. Do you think it might be more appropriate here?

Reply:
That looks like a good reference.  No harm in having both if both are relevant.

Reply:
I will keep both then.